### ISSUE TITLE
Override framework controls with app_includes

### ISSUE THREAD

**Issue by kmott** [qcubed/qcubed#677]
Hello,

I'm trying to override QControlBase.class.php by using app_includes, but it doesn't seem to be working.  I've copied by customized QControlBase.class.php (with an additional property with corresponding __set() and __get() entry) to app_includes/core/QControlBase.class.php.  My entry in app_includes/app_includes.inc.php looks like this:

``` php
  QApplicationBase::$ClassFile['qcontrolbase'] = __APP_INCLUDES__ . '/core/QControlBase.class.php';
```

When I then instantiate a control (for example QLinkButton()), and call my new property, it just throws the standard "Undefined SET property or variable in 'QLinkButton' class: < propertyname >".

The error stack trace has the incorrect QControlBase.class.php loaded from "qcubed/_core/base_controls/QControlBase.class.php".  If I inspect the QApplicationBase::$ClassFile array, it does have the correct class info listed:

``` php
['qcontrolbase'] => '/path/to/includes/app_includes/core/QControlBase.class.php'
```

But, it's almost as if the Form has already loaded the QControlBase from base_controls, and ignores me telling it to use the one from app_includes.

Has anybody else run into this, or have a suggestion on how I can workaround it?

**Comment by vaibhav-kaushal** [comment]
Look here: 
https://github.com/qcubed/framework/blob/master/includes/qcubed/_core/framework/QApplicationBase.class.php#L391

and here: https://github.com/qcubed/framework/blob/master/includes/qcubed/_core/qcubed.inc.php#L68

You cannot override the *preloaded* classes using app_includes.

**Comment by kmott** [comment]
Then we should probably update the hint-text in app_includes.inc.php, as it pretty clearly uses QControlBase.class.php as the example to override a core framework control.

```php
	 /**
	 * ===============================
	 * MODIFYING QCUBED CORE FILES
	 * ===============================
	 *
	 * This file is intended to be used to override most core files.
	 * For example, if you want to override the QControlBase class then you should do the following:
	 * -------------
	 * 1. Copy the QControlBase.class.php file and paste it somewhere in the __APP_INCLUDES__ directory
	 *    (i.e the directory where this file is located)
	 *    Let us say you paste it in 'core_overrides' directory under __APP_INCLUDES__.
	 *
	 * 2. Make your desired modifications in the newly pasted file and save it.
	 *
	 * 3. Add the line
	 *      QApplicationBase::$ClassFile['qcontrolbase'] = __APP_INCLUDES__ .'/core_overrides/QControlBase.class.php';
	 *    to this file (app_includes.inc.php) at the bottom
	 *
	 * 4. Save this file (app_includes.inc.php).
	 */
```

**Comment by vaibhav-kaushal** [comment]
Thanks for noticing that @kmott. Please send a pull request for that.

**Comment by spekary** [comment]
QControlBase is not editable. You should be editing the QControl.class.php file. That file overrides QControlBase.class.php.

**Comment by spekary** [comment]
Fixed in #1220
